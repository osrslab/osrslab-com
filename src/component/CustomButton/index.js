import React from 'react';
import PropTypes from 'prop-types';

/*
  CustomButton Component
  @param {string} name
  @param {func} handleClick
  @returns button element
 */
const CustomButton = ({ name, handleClick }) => (
    <input className="button" type="button" value={name} onClick={handleClick} />
);

CustomButton.propTypes = {
  name: PropTypes.string.isRequired,
  handleClick: PropTypes.func.isRequired,
};

export default CustomButton;
