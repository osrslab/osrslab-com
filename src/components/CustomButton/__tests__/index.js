import React from 'react';
import { shallow } from 'enzyme';
import CustomButton from '../index';

const handleClick = () => 'hello world';

describe('<CustomButton />', () => {
  it('should be defined', () => {
    expect(CustomButton).toBeDefined();
  });

  it('should render correctly', () => {
    const tree = shallow(
      <CustomButton name='button test' handleClick={handleClick} />,
    );
    expect(tree).toMatchSnapshot();
  });
});
