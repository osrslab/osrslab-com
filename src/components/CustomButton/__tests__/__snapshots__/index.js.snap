// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CustomButton /> should render correctly 1`] = `
<input
  className="button"
  onClick={[Function]}
  type="button"
  value="button test"
/>
`;
